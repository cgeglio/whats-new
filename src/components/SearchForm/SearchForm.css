.searchForm {
  grid-area: search;
  height: 80px;
}
