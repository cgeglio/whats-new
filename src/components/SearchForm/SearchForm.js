import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {keyword: ''}
  }

  updateKeyword = event => {
    this.setState({keyword: event.target.value})
  }

  findArticles = event => {
    event.preventDefault();
    let articles = this.props.newsItems.reduce((acc, n) => {
      n.articles.forEach(a => acc.push(a))
      return acc;
    }, []);
    this.updateArticles(articles);
  }

  updateArticles = articles => {
    this.props.updateFromSearch({topic: this.state.keyword, articles: articles.filter(a => a.headline.toUpperCase().includes(this.state.keyword.toUpperCase()))})
    this.setState({keyword: ''})
  }

  render() {
    return (
      <form>
        <input
          type="text"
          placeholder="Search for articles..."
          name="keyword"
          value={this.state.keyword}
          onChange={this.updateKeyword}
        />
        <button className="search" onClick={this.findArticles}>Search</button>
      </form>
    )
  }
}

export default SearchForm;
