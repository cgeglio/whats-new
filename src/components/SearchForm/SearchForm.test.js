import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<SearchForm
      newsItems={[
        {topic: "local", articles: []},
        {topic: "entertainment", articles: []}
      ]}
      updateFromSearch={jest.fn()}
    />);
  expect(wrapper).toMatchSnapshot();
  });

  it('should update state with a new keyword when updateKeyword is called', () => {
    const wrapper = shallow(<SearchForm />);
    const mockEvent = {target: {name: 'keyword', value: 'denver'}};
    const expected = {keyword:'denver'};

    expect(wrapper.state()).toEqual({keyword: ''});
    wrapper.instance().updateKeyword(mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });

  it('should find search related articles and call updateArticles when findArticles is called', () => {
    const wrapper = shallow(<SearchForm
      newsItems={[
        {topic: "local", articles: []},
        {topic: "entertainment", articles: []}
      ]}
      updateFromSearch={jest.fn()}
    />);
    const mockEvent = {preventDefault: jest.fn()};

    wrapper.instance().updateArticles = jest.fn();
    wrapper.instance().findArticles(mockEvent);
    expect(wrapper.instance().updateArticles).toHaveBeenCalled();
  });

  it('should call updateFromSearch and reset state when updateArticles is called', () => {
    const wrapper = shallow(<SearchForm
      updateFromSearch={jest.fn()}
    />);
    const mockArticles = [{headline: 'denver'}, {headline: 'colorado'}]
    const expected = {keyword: ''};

    wrapper.instance().updateArticles(mockArticles);
    expect(wrapper.instance().props.updateFromSearch).toHaveBeenCalled();
    expect(wrapper.state()).toEqual(expected);
  });
});
