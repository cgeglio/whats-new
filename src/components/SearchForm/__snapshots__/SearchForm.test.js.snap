// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with all data passed in correctly 1`] = `
<form>
  <input
    name="keyword"
    onChange={[Function]}
    placeholder="Search for articles..."
    type="text"
    value=""
  />
  <button
    className="search"
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
